src/FilesystemRepository.php

```diff
@@ -254,6 +254,35 @@ public function remove($query, $language = 'glob')
     /**
      * {@inheritdoc}
      */
+    public function move($sourceQuery, $targetPath, $language = 'glob')
+    {
+        $iterator = $this->getGlobIterator($sourceQuery, $language);
+        $moved = 0;
+
+        Assert::notEq('', trim($sourceQuery, '/'), 'The root directory cannot be moved.');
+        Assert::stringNotEmpty($targetPath, 'The target path must be a non-empty string. Got: %s');
+        Assert::startsWith($targetPath, '/', 'The target path %s is not absolute.');
+
+        $targetPath = Path::canonicalize($targetPath);
+
+        $sourcePaths = iterator_to_array($iterator);
+
+        if (1 === count($sourcePaths) && is_file($sourcePaths[0])) {
+            $this->moveResource($sourcePaths[0], $targetPath, $moved);
+        } else {
+            $this->ensureDirectoryExists($targetPath);
+
+            foreach ($sourcePaths as $sourcePath) {
+                $this->moveResource($sourcePath, $targetPath.'/'.Path::getFilename($sourcePath), $moved);
+            }
+        }
+
+        return $moved;
```

Comment:
@WouterJ should this be declared in the documentation? Is this meant to be part of the API?